scripts/validate.sh

```diff
@@ -36,11 +36,29 @@ validate_file_nonexists '/var/log/secure'
 validate_file_nonexists '/var/log/wtmp'
 
 actual_kernel=$(uname -r)
-echo "Verifying that kernel version $actual_kernel matches $KERNEL_VERSION"
+echo "Verifying that kernel version $actual_kernel matches $KERNEL_VERSION..."
 
 if [[ $actual_kernel == $KERNEL_VERSION* ]]; then
-  echo "Kernel matches expected version"
+  echo "Kernel matches expected version!"
 else
-  echo "Kernel does not match expected version."
+  echo "Kernel does not match expected version!"
   exit 1
 fi
+
+echo "Verifying that the kernel has the correct versionlock..."
+
+# only one version of the kernel should be version locked
```

Comment:
Should we generalize this to other packages too? I can't think of a reason why we'd versionlock multiple versions of the same package.

Reply:
Yeah, I considered that. The format of the versionlock file isn't particularly conducive to that sort of check, though it's not impossible.

The format is `EPOCH:NAME-VERSION-RELEASE.ARCH`, more here: https://man7.org/linux/man-pages/man1/yum-versionlock.1.html
```
0:kernel-4.14.304-226.531.amzn2.*
0:kernel-5.10.165-143.735.amzn2.*
0:runc-1.1.4-1.amzn2.*
0:containerd-1.6.6-1.amzn2.0.2.*
```

We need to convert these entries to just a list of `NAME`-s. Packages often use `-` as a delimiter in their names, so we can't just split on that (e.g. `nvidia-driver-latest-dkms`). We could chop off the `EPOCH` and query the index, like:
```
> rpm -q 'kernel-4.14.304-226.531.amzn2.*' --queryformat '%{NAME}\n'
kernel
```

But this won't detect the issue we saw in 20220211, because the `kernel-4.14` package was no longer installed:
```
> rpm -q kernel-4.14
package kernel-4.14 is not installed
```

We'll need to delete any versionlocks for packages that aren't currently installed, *then* do this check on the uniqueness of the package names. I'll put up a revision.